catroid/src/phiro/java/org/catrobat/catroid/ui/fragment/FormulaEditorCategoryListFragment.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.ui.fragment;
+
+import android.content.Context;
+import android.os.Bundle;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.SettingsActivity;
+
+public class FormulaEditorCategoryListFragment extends MainFormulaEditorCategoryListFragment {
+
+	@Override
+	public void onCreate(Bundle savedInstanceState) {
+		initCategories();
+		super.onCreate(savedInstanceState);
+	}
+
+	private void initCategories() {
+		initSensors();
```

Comment:
I've refactored that now so that the code is even more maintainable using the template method pattern. 

This means MainFormulaEditorCategoryListFragment knows the "recipe" of which categories to setup. The "recipe" is initCategories:

	private void initCategories() {
		initObjects();
		initFunctions();
		initLogic();
		initSensors();
	}
These init-methods are protected, so that the subclass can override it to setup the categories with help of polymorphism. 

So, all that needs to be done now in the "Phiro-"FormulaEditorCategoryListFragment is override the method initSensors() and call super.initSensors()

Even much shorter and better maintainable.